heralding/capabilities/rdp.py

```diff
@@ -0,0 +1,171 @@
+
+        await self.send_data(writer, mcs_cres, tls_obj)
+        logger.debug("Sent MCS Connect Response PDU")
+
+        if tls_obj:
+            data = await self.recv_data(reader, 41, tls_obj)
+        else:
+            data = await self.recv_data(reader, 512, tls_obj)
+
+        if not data:
+            logger.debug("Expected ErectDomainRequest. Got Nothing.")
+            return
+        if not ErectDomainRequestPDU.checkPDU(data):
+            logger.debug("Malformed Packet Received. Expected ErectDomainRequest.")
+            session.end_session()
+            return
+
+        # in tls attach user request and erect domain are not merged together
+        logger.debug("Received: ErectDomainRequest and AttactUserRequest(not in tls) : "+repr(data))
+
+        er_len = len(data)
+        # if we got only erectdomian req in previous read, len should be less than 13
+        if er_len < 13:
+            # Here , this dosen't occur in rdp_security mode, but it tls mode it keeps waiting for data
+            data = await self.recv_data(reader, 64, tls_obj)
+            # # TODO: check if attach user req
+            logger.debug("Received: Attach USER req : "+repr(data))
+
+        mcs_usrcnf = MCSAttachUserConfirmPDU().getFullPacket()
+        await self.send_data(writer, mcs_usrcnf, tls_obj)
+        logger.debug("Sent: Attach User Confirm")
+
+        # Handle multiple Channel Join request PUDs
+        for _ in range(7):
+            # data = await reader.read(2048)
+            data = await self.recv_data(reader, 2048, tls_obj)
+            if not data:
+                logger.debug("Expected: Channel Join/Client Security Packet.Got Nothing.")
+                return
+            channel_req = MCSChannelJoinRequestPDU()
+            v = channel_req.parse(data)
+            if v < 0:
+                break
+            channel_id = channel_req.channelID
+            channel_init = channel_req.initiator
+            channel_cnf = MCSChannelJoinConfirmPDU(channel_init, channel_id).getFullPacket()
+
+            await self.send_data(writer, channel_cnf, tls_obj)
+            logger.debug("Sent: MCS Channel Join Confirm of channel %s" % (channel_id))
+
+        # Handle Client Security Exchange PDU
+        if not data:
+            # data = await reader.read(2048)
+            data = await self.recv_data(reader, 2048, tls_obj)
+
+        # There is no client security exchange in TLS Security
+        if tls_obj:
+            client_info = ClientInfoPDU()
+            client_info.parseTLS(data)
+            username = client_info.rdpUsername
+            password = client_info.rdpPassword
+            session.add_auth_attempt('plaintext', username=username, password=password)
+
+        else:
+            client_sec = ClientSecurityExcahngePDU()
+            client_sec.parse(data)
+            self.encClientRandom = client_sec.encClientRandom
+            print("ENC_CLIENT_RANDOM: ", self.encClientRandom)
```

Comment:
Please remove all print statements and also other code that's for _development purpose_.

Reply:
okay